examples/ml_pipeline.md

```diff
@@ -0,0 +1,208 @@
+```
+
+## Run Maximum Common Substructure Search
+To figure out which Fragalysis structures to dock each ligand to, we run an MCS search
+for all ligands against all the crystal structure ligands from Fragalysis. This will
+rank all the structures based on the similarity between the crystal ligand and each
+ligand to be docked. During the docking step we will determine how many of the top
+structures each ligand will actually be docked to.
+
+This example script call will use the downloaded CDD data and the downloaded and
+extracted Fragalysis data from the previous steps. We will also run this step with 32
+parallel processes. This number can be adjusted depending on your machine, but it is
+recommended to parallelize this script as it can take a while otherwise.
+```bash
+mkdir -p mcs_results
+python /path/to/docking/scripts/find_all_mcs.py \
+    -exp cdd_achiral_enantiopure_semiquant_schema.json \
+    -x ./mpro_fragalysis/extra_files/Mpro_compound_tracker_csv.csv \
+    -x_dir ./mpro_fragalysis/aligned/ \
+    -o ./mcs_results/ \
+    -n 32 \
+    -n_draw 0
+```
+The `-n_draw` option tells the script how many of the top matches for each compound to
+draw the structure of, with the matched substructure highlighted. This can be useful if
+you want to inspect how the MCS algorithm is working, but for production is not
+necessary, so we use `-n_draw 0`.
+
+## Prep OpenEye Design Units
+This script takes advantage of the new `click` CLI, so it can only be called using the
+CLI command `asap-prep`. In the future, the other steps in this pipeline will be updated
+to follow this same pattern.
+
+This example will prep every structure downloaded in Fragalysis
+(`--fragalysis-dir ./mpro_fragalysis/`), and will store each individual prepped
+structure as both a `.oedu` file and a JSON file
+(`--gen-cache structures --cache-type DesignUnit --cache-type JSON`). We also use `dask`
+to parallelize the job locally (`--use-dask --dask-type local`). We will also use the
+RCSB OpenEye loop database (`--loop-db ./rcsb_spruce.loop_db`), which will need to be
+downloaded separately.
+```bash
+mkdir -p prepped_receptors
+asap-prep protein-prep \
+    --target SARS-CoV-2-Mpro \
+    --loop-db ./rcsb_spruce.loop_db \
+    --fragalysis-dir ./mpro_fragalysis/ \
+    --gen-cache structures \
+    --cache-type DesignUnit \
+    --cache-type JSON \
+    --use-dask \
+    --dask-type local \
+    --output-dir ./prepped_receptors
+```
+
+## Run Docking
+The final step in preparing the structural data is to run docking. There are a number of
+options that can be configured, but this a minimal example that will dock each compound
+to its top match from the MCS search. This step is again parallelized (`-n 32`), which
+can be adjusted based on your machine. Note that the single quotes in the `-r` option
+are required to prevent the shell from expanding the wildcard.
+```bash
+mkdir -p docking_results
+python /path/to/docking/scripts/run_docking_oe.py \
```

Comment:
Can we make this step use new docking or is it to convoluted atm?

Reply:
I think this'll need the large scale docking since it's for the whole CDD. I didn't think we had that updated yet, but if we do I'm happy to rewrite to use it

Reply:
We do have large scale docking ready but they are probably not compatible if relying on file names etc. Good enough for now and we can iterate.